File: docs-examples/example-java/src/test/java/io/pollend/quartz/docs/scheduler/SampleScheduler.java
package io.pollend.quartz.docs.scheduler;

// tag::imports[]
import io.pollend.quartz.annotation.QuartzKey;
import io.pollend.quartz.annotation.ScheduleOn;
import org.quartz.JobKey;
import org.quartz.Trigger;
// end::imports[]

// tag::clazz[]
public interface SampleScheduler {
    @ScheduleOn(MessageJob1.class)
    void messageWithoutKey(@QuartzKey("message") String value);

    @ScheduleOn(MessageJob1.class)
    void messageJobWithKey(JobKey key, @QuartzKey("message") String value);

    @ScheduleOn(MessageJob1.class)
    void messageJobWithTrigger(JobKey key, Trigger trigger, @QuartzKey("message") String value);
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/pollend/quartz/docs/scheduler/SampleScheduler.kt
package io.pollend.quartz.docs.scheduler

// tag::imports[]
import io.pollend.quartz.annotation.QuartzKey
import io.pollend.quartz.annotation.ScheduleOn
import org.quartz.JobKey
import org.quartz.Trigger
// end::imports[]

// tag::clazz[]
interface SampleScheduler {

    @ScheduleOn(MessageJob1::class)
    fun messageWithoutKey(@QuartzKey("message") value: String?)

    @ScheduleOn(MessageJob1::class)
    fun messageJobWithKey(key: JobKey?, @QuartzKey("message") value: String?)

    @ScheduleOn(MessageJob1::class)
    fun messageJobWithTrigger(key: JobKey?, trigger: Trigger?, @QuartzKey("message") value: String?)
}
// end::clazz[]
